storage/client/file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -3,77 +3,484 @@
+     * {@codesnippet com.azure.storage.file.directoryAsyncClient.create#map}
+     *
+     * @param metadata Optional. Metadata to associate with the directory
+     * @return A response containing the directory info and the status of creating the directory.
+     * @throws StorageErrorException If the directory has already existed, the parent directory does not exist or directory name is an invalid resource name.
+     */
     public Mono<Response<DirectoryInfo>> create(Map<String, String> metadata) {
-        throw new UnsupportedOperationException();
+        return azureFileStorageClient.directorys().createWithRestResponseAsync(shareName, directoryName, null, metadata, Context.NONE)
+            .map(this::createWithRestResponse);
     }
 
+    /**
+     * Deletes the directory in the storage account.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Delete the directory</p>
+     *
+     * {@codesnippet com.azure.storage.file.directoryClient.delete}
+     *
+     * @return A response that only contains headers and response status code
+     * @throws StorageErrorException If the share doesn't exist
+     */
     public Mono<VoidResponse> delete() {
-        throw new UnsupportedOperationException();
+        return azureFileStorageClient.directorys().deleteWithRestResponseAsync(shareName, directoryName, Context.NONE).map(VoidResponse::new)
+            .map(VoidResponse::new);
+    }
+
+    /**
+     * Retrieves the properties of the storage account's directory.
+     * The properties includes directory metadata, last modified date, is server encrypted, and eTag.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Retrieve directory properties</p>
+     *
+     * {@codesnippet com.azure.storage.file.directoryAsyncClient.getProperties}
+     *
+     * @return Storage directory properties
+     */
+    public Mono<Response<DirectoryProperties>> getProperties() {
+        return azureFileStorageClient.directorys().getPropertiesWithRestResponseAsync(shareName, directoryName, snapshot, null, Context.NONE)
+            .map(this::getPropertiesResponse);
     }
 
-    public Mono<Response<DirectoryProperties>> getProperties(String shareSnapshot) {
-        throw new UnsupportedOperationException();
+    /**
+     * Sets the user-defined metadata to associate to the directory.
+     *
+     * <p>If {@code null} is passed for the metadata it will clear the metadata associated to the directory.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Set the metadata to "directory:updatedMetadata"</p>
+     *
+     * {@codesnippet com.azure.storage.file.directoryAsyncClient.setMetadata#map}
+     *
+     * <p>Clear the metadata of the directory</p>
+     *
+     * {@codesnippet com.azure.storage.file.directoryAsyncClient.setMetadata#map.clearMetadata}
+     *
+     * @param metadata Optional. Metadata to set on the directory, if null is passed the metadata for the directory is cleared
```

Comment:
Don't write these as "Optional. Metadata ..." - make it a full sentence, e.g. "Optional metadata to set on the directory..".
Do this in all cases where the "Optional. " approach is being used currently.

Reply:
Done.